### ISSUE TITLE
Empty event data is delivered with the knative integrated eventing to lambda

### ISSUE THREAD

**Issue by abbi-gaurav** [kyma-project/kyma#3070]
**Description**
On testing the [e2e-scenario](https://github.com/kyma-incubator/examples/tree/master/kyma-local-e2e-scenario), it was discovered that the event delivered to the lambda is an empty event.

```
{ 'event-type': undefined,
  'event-id': undefined,
  'event-time': undefined,
  'event-namespace': undefined,
  data: {},
  extensions:
   { request:
      IncomingMessage {
        _readableState: [Object],
        readable: true,
        domain: null,
        _events: {},
        _eventsCount: 0,
        _maxListeners: undefined,
        socket: [Object],
        connection: [Object],
        httpVersionMajor: 1,
        httpVersionMinor: 1,
        httpVersion: '1.1',
        complete: false,
        headers: [Object],
        rawHeaders: [Array],
        trailers: {},
        rawTrailers: [],
        aborted: false,
        upgrade: false,
        url: '/',
        method: 'POST',
        statusCode: null,
        statusMessage: null,
        client: [Object],
        _consuming: false,
        _dumped: false,
        next: [Function: next],
        baseUrl: '',
        originalUrl: '/',
        _parsedUrl: [Object],
        params: [Object],
        query: {},
        res: [Object],
        _startAt: [Array],
        _startTime: 2019-03-08T12:29:50.514Z,
        _remoteAddress: '::ffff:127.0.0.1',
        body: {},
        route: [Object] },
     response:
      ServerResponse {
        domain: null,
        _events: [Object],
        _eventsCount: 2,
        _maxListeners: undefined,
        output: [],
        outputEncodings: [],
        outputCallbacks: [],
        outputSize: 0,
        writable: true,
        _last: false,
        upgrading: false,
        chunkedEncoding: false,
        shouldKeepAlive: true,
        useChunkedEncodingByDefault: true,
        sendDate: true,
        _removedConnection: false,
        _removedContLen: false,
        _removedTE: false,
        _contentLength: null,
        _hasBody: true,
        _trailer: '',
        finished: false,
        _headerSent: false,
        socket: [Object],
        connection: [Object],
        _header: null,
        _onPendingData: [Function: bound updateOutgoingData],
        _sent100: false,
        _expect_continue: false,
        req: [Object],
        locals: {},
        _startAt: undefined,
        _startTime: undefined,
        writeHead: [Function: writeHead],
        __onFinished: [Object],
        [Symbol(outHeadersKey)]: [Object] } } }```

```

The event content is somewhat getting lost between
publish --> nats-dispatcher (storing) --> nats streaming --> dispacher (event delivery).

**AC**
- [ ] Correct event payload is delivered to lambda and 
- [ ] Payload is contained in the `event.data` part of JSON in the lambda.

**Comment by radufa** [comment]
Installed as a subscriber an "echo" service which dumps the incoming request and tested with and without KN:
1 POST to publish pod:
```
curl -i -H "Content-Type:application/json" -X POST http://localhost:8080/v1/events -d '{"source-id": "external-application", "event-type": "test-event-bus", "event-type-version": "v1", "event-time": "2018-11-02T22:08:41+00:00", "data": {"event":{"customer":{"customerID": "1234", "uid": "rick.sanchez@mail.com"}}}}'
```

2. With KN, echo subscriber log:
```
2019/03/09 18:46:44 "POST /echo HTTP/1.1\r\nHost: 192.168.1.56:31380\r\nAccept-Encoding: gzip\r\nContent-Length: 432\r\nUser-Agent: Go-http-client/1.1\r\nX-B3-Sampled: 1\r\nX-B3-Spanid: 3e970cb5ea254ac4\r\nX-B3-Traceid: 3e970cb5ea254ac4\r\nX-Envoy-Internal: true\r\nX-Forwarded-For: 172.17.0.1\r\nX-Forwarded-Proto: http\r\nX-Request-Id: b2b45a5e-f8c0-938d-9fae-5986cee964aa\r\n\r\n{\"{\"source-id\":\"external-application\",\"event-type\":\"test-event-bus\",\"event-type-version\":\"v1\",\"event-id\":\"9c978589-b0f0-4de0-a925-b28102a9b6be\",\"event-time\":\"2018-11-02T22:08:41+00:00\",\"data\":{\"event\":{\"customer\":{\"customerID\":\"1234\",\"uid\":\"rick.sanchez@mail.com\"}}},\"SourceIdFromHeader\":false,\"extensions\":{\"trace-context\":{\"x-b3-flags\":\"0\",\"x-b3-sampled\":\"1\",\"x-b3-spanid\":\"4577bca8e7af7301\",\"x-b3-traceid\":\"1db19aa82d3f5961\"}}}\"}"
````

3. Without KN, echo subscriber log:
```
2019/03/10 09:53:51 "POST /echo HTTP/1.1\r\nHost: 192.168.1.56:31380\r\nContent-Length: 74\r\nContent-Type: application/json\r\nKyma-Event-Id: 5f249c8d-ab4b-4b47-a973-3b5f118cd8a1\r\nKyma-Event-Time: 2018-11-02T22:08:41+00:00\r\nKyma-Event-Type: test-event-bus\r\nKyma-Event-Type-Version: v1\r\nKyma-Source-Id: external-application\r\nKyma-Subscription: example-subscription\r\nUser-Agent: Go-http-client/1.1\r\nX-B3-Flags: 0\r\nX-B3-Parentspanid: 7072330b32b41917\r\nX-B3-Sampled: 1\r\nX-B3-Spanid: 7aa84bd79d24fed2\r\nX-B3-Traceid: 5877867119168241\r\nX-Envoy-Internal: true\r\nX-Forwarded-For: 172.17.0.1\r\nX-Forwarded-Proto: http\r\nX-Request-Id: 4a0664e4-a4a6-97da-93c7-8bc45e8155c2\r\n\r\n{\"event\":{\"customer\":{\"customerID\":\"1234\",\"uid\":\"rick.sanchez@mail.com\"}}}"
```
Comparing, KN sends a CE event to the subscriber:

{\"{\"source-id\":\"external-application\",\"event-type\":\"test-event-bus\",\"event-type-version\":\"v1\",\"event-id\":\"9c978589-b0f0-4de0-a925-b28102a9b6be\",\"event-time\":\"2018-11-02T22:08:41+00:00\",\"data\":{\"event\":{\"customer\":{\"customerID\":\"1234\",\"uid\":\"rick.sanchez@mail.com\"}}},\"SourceIdFromHeader\":false,\"extensions\":{\"trace-context\":{\"x-b3-flags\":\"0\",\"x-b3-sampled\":\"1\",\"x-b3-spanid\":\"4577bca8e7af7301\",\"x-b3-traceid\":\"1db19aa82d3f5961\"}}}\"}"

but not Event-Bus push:
{\"event\":{\"customer\":{\"customerID\":\"1234\",\"uid\":\"rick.sanchez@mail.com\"}}}"

which sends only the content of "data".

**Comment by abbi-gaurav** [comment]
Observed that it seems we are adding extra `{}` to the JSON as a string. This is visible in the payload `{"{"source-id"` . (double `{{`).
This could be one reason that kubeless is failing.

As discussed, this needs to be fixed.
After that, we need to check the eventing scenario with lambda example.
https://github.com/kyma-project/examples/tree/master/event-subscription/lambda.

Kubeless already supports cloud events. https://kubeless.io/docs/pubsub-functions/

**Comment by abbi-gaurav** [comment]
Fix will be done in #3057.
